public_html/wp-content/mu-plugins/camptix-tweaks/addons/ticket-types/ticket-types.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * Add a fixed set of ticket types to every site, globally defined.
+ */
+
+namespace WordCamp\CampTix_Tweaks\Ticket_Types;
+defined( 'WPINC' ) || die();
+
+const META_KEY = 'tix_type';
+
+require_once __DIR__ . '/remote.php';
+
+add_action( 'camptix_init', __NAMESPACE__ . '\camptix_init' );
+
+/**
+ * Hook into WordPress and Camptix.
+ */
+function camptix_init() {
+	register_post_meta(
+		'tix_ticket',
+		META_KEY,
+		array(
+			'single'            => true,
+			'sanitize_callback' => __NAMESPACE__ . '\get_slug_from_value',
+		)
+	);
+
+	add_action( 'camptix_add_meta_boxes', __NAMESPACE__ . '\add_types_meta_box' );
+	add_action( 'save_post', __NAMESPACE__ . '\save_post' );
+
+	add_filter( 'manage_edit-tix_ticket_columns', __NAMESPACE__ . '\manage_columns_filter', 9 );
+	add_action( 'manage_tix_ticket_posts_custom_column', __NAMESPACE__ . '\manage_columns_output', 10, 2 );
+}
+
+/**
+ * Get the list of ticket types.
+ *
+ * @return array[] List of associative arrays representing ticket-types.
+ */
+function get_types() {
+	return apply_filters(
+		'camptix_ticket_types',
+		array(
+			array(
+				'slug' => 'default',
+				'name' => __( 'Default', 'wordcamporg' ),
+				'description' => __( 'Default, in-person attendee.', 'wordcamporg' ),
+			),
+		)
+	);
+}
+
+/**
+ * Match the given value to an existing type slug, return the slug, or "default" if not a match.
+ */
+function get_slug_from_value( $value ) {
+	$match = wp_list_pluck( wp_list_filter( get_types(), array( 'slug' => $value ) ), 'slug' );
+	if ( count( $match ) > 0 ) {
+		return current( $match );
+	}
+	return 'default';
+}
```

Comment:
What if the types were registered as an associative array, with the slug as the key? Then this function might be simpler or even unnecessary.

Reply:
Good idea — less magic with `wp_list_*` 🙂